### ISSUE TITLE
Ubuntu 18.04 Root on ZFS should mirror boot

### ISSUE THREAD

**Issue by erik-wramner** [zfsonlinux/zfs#8223]
### Describe the problem you're observing

@rlaager this is a feature request for the excellent howto https://github.com/zfsonlinux/zfs/wiki/Ubuntu-18.04-Root-on-ZFS. It covers both mirrored and single disk setups. Personally I found it a little bit hard to follow with mirroring as I was unsure about some steps. Clearer instructions about when to do something on all disks and when to do it only one one would make the guide even better. However, the only real problem I found was that the boot partition for LUKS installs seems to be only on one disk? At least mine ended up that way. That means there is a single point of failure. Ideally the boot partition should use mdadm or something to be fault tolerant.

### Describe how to reproduce the problem
Follow the howto with two or more disks. When done disconnect the disk that got the boot partition and try to boot.

**Comment by erik-wramner** [comment]
I'm sorry, but I don't follow. How would that help making the boot partition (step 3.4, /dev/disk/by-id/scsi-SATA_disk1-part4) mirrored to the other disks? Do you mean that DilOS supports that out of the box or doesn't need it? Unfortunately I can only pick Red Hat/CentOS or Ubuntu, which means I'm stuck with Ubuntu and LUKS for ZFS, so at least for now I do need a mirrored boot partition to tolerate the failure of a single drive.

**Comment by ikozhukhov** [comment]
DilOS is illumos based platform, and yes - support mirrored boot.
we are using old grub1, but with new 'loader' we can boot from raidz
i pointed to DilOS as reference for updates in APT/DPKG for install packages to another root.
for example: 
you can create new BE:
`beadm create new-dilos`
mount it to another root:
`beadm mount new-dilos /a`
install package to another root:
`apt -R /a install new-package`
activate BE for next boot with new package:
`beadm activate new-dilos`
reboot to `new-dilos` BE

**Comment by rlaager** [comment]
I intend to make some changes to the HOWTO along these lines. This was recently discussed on zfs-discuss:
http://list.zfsonlinux.org/pipermail/zfs-discuss/2018-December/032911.html
http://list.zfsonlinux.org/pipermail/zfs-discuss/2018-November/032679.html
http://list.zfsonlinux.org/pipermail/zfs-discuss/2018-December/032792.html

**Comment by erik-wramner** [comment]
Right, sounds good. Thanks!

**Comment by rlaager** [comment]
I'm pushing changes now. I didn't re-test the mirror scenarios, though I did improve the instructions. If you or someone else wants to retest them and let me know if there are any bugs, that'd be helpful. If not, eventually someone will try to use them. :)

**Comment by erik-wramner** [comment]
I'll follow the instructions as soon as I get some spare time. Not sure when that happens, so perhaps someone else will beat me to it.

**Comment by erik-wramner** [comment]
@rlaager I finally found time to repeat the installation. This time it worked without a hitch in a mirrored setup with LUKS. You could perhaps add a note to "4.8b Install GRUB for UEFI booting" that says something like "Note that the partition will be cloned later in mirrored configurations, so only one disk needs to be configured here". I wondered if the instructions were correct and had to skip ahead to be sure. They were, so no problem really!

**Comment by rlaager** [comment]
@erik-wramner Thanks for the suggestion. I've added a note to the 18.04 HOWTO as well as the Debian versions.
